File: part1-coding/04-intervals-merge/java/04IntervalsMerge.java
import java.util.*;
public class MergeIntervals {
    public static int[][] merge(int[][] in){
        Arrays.sort(in,(a,b)->a[0]-b[0]);
        List<int[]> res=new ArrayList<>();
        for(int[] cur: in){
            if(res.isEmpty() || res.get(res.size()-1)[1] < cur[0]) res.add(new int[]{cur[0],cur[1]});
            else res.get(res.size()-1)[1] = Math.max(res.get(res.size()-1)[1], cur[1]);
        }
        return res.toArray(new int[res.size()][]);
    }
}

File: part1-coding/04-intervals-merge/cpp/04-intervals-merge.cpp
#include <bits/stdc++.h>
using namespace std;
vector<vector<int>> merge(vector<vector<int>>& in){
    sort(in.begin(), in.end(), [](auto&a,auto&b){return a[0]<b[0];});
    vector<vector<int>> res;
    for(auto &c: in){
        if(res.empty() || res.back()[1] < c[0]) res.push_back(c);
        else res.back()[1] = max(res.back()[1], c[1]);
    } return res;
}
int main(){ return 0; }
